doc/source/ray-contribute/ci.rst

```diff
@@ -0,0 +1,50 @@
+CI Testing Workflow on PRs
+==========================
+
+This guide helps contributors to understand the CI workflow on a PR.
+
+microcheck: default tests on your PR
+------------------------------------
+With every commit on your PR, by default, we'll run a set of tests
+called `microcheck`.
+
+These tests are designed to be 90% accurate at catching bugs on your
+PR while running only 10% of the full test suite. As a result,
+microcheck typically finishes twice as fast and twice cheaper than
+the full test suite. Some of the notable features of microcheck are:
+
+* If a new test is added or an existing test is modified in a pull
+  request, microcheck will ensure these tests are included.
+* You can manually add more tests to microcheck by including the following line
+  in the body of your git commit message:
+  `@microcheck TEST_TARGET01 TEST_TARGET02 ....`. This line must be in the
+  body of your message, starting from the second line or
+  below (the first line is the commit message title). For example, here
+  is how I manually add tests in my pull request::
+
+    // git command to add commit message
+    git commit -a -s
```

Comment:
maybe don't add `-a`? probably should not encourage / tell user to use `-a` here as if `-a` is required for the workflow.

Reply:
what is your favorite workflow to add a git commit message? the `-a` is required to add a new commit message (there are similar alternatives like `-m` but they will need something)